routers/web/auth/oauth.go

```diff
@@ -470,9 +470,9 @@ func AuthorizeOAuth(ctx *context.Context) {
 		return
 	}
 
-	// Redirect if user already granted access and the application is confidential.
-	// I.e. always require authorization for public clients as recommended by RFC 6749 Section 10.2
-	if app.ConfidentialClient && grant != nil {
+	// Redirect if user already granted access and the application is confidential or trusted otherwise
+	// I.e. always require authorization for untrusted public clients as recommended by RFC 6749 Section 10.2
+	if (app.ConfidentialClient || app.SkipSecondaryAuthorization) && grant != nil {
```

Comment:
This code reads like it should be
```suggestion
	if (app.ConfidentialClient || !app.SkipSecondaryAuthorization) && grant != nil {
```
or am I missing something?
It looks as if you say `if skip authorization then ask user to authorize themselves` right now.

Reply:
No, basically the code says:

`if access already granted AND (confidential client OR skip secondary auth) then use the existing grant to authorize the application directly.